### PR TITLE
Files for stm32h7zi2

### PR DESCRIPTION
**Pull request by robertobucher**
## Summary
Files for using pysimCoder with the NUCLEO-H743ZI2 board

### CODE REVIEW

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include "chip.h"
+#include "arm_internal.h"
+#include "stm32_pwm.h"
+#include "nucleo-h743zi2.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define OK 0
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c:42]: maybe better to add proper include?

File: boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h [hunk 2]
```diff
@@ -148,4 +201,20 @@ void weak_function stm32_usbinitialize(void);
...
+ *   Initialize PWM and register the PWM device.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_PWM
+int stm32_pwm_setup(void);
+#endif
+
+#ifdef CONFIG_SENSORS_QENCODER
+int stm32h7_qencoder_initialize(const char *devpath, int timer);
```
[the remark this answers is not in the document]
Comment (acassis) [reply boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h:217]: @pkarashchenko although I agree with your suggestion, I think we are doing it incorrectly: see, all arch:stm32/stm32f7/stm32h7, etc are using "stm32_" for function names. It makes it hard to IDE find the right definition. I think we should always adopt the arch name instead using a same stm32_ prefix for all of them.

[the remark this answers is not in the document]
Comment (pkarashchenko) [reply boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h:217]: yes. I just want to make things consistent. Either all use `stm32h7_` or `stm32_`

File: arch/arm/src/stm32h7/stm32_adc.c [hunk 3]
```diff
@@ -767,7 +767,7 @@ static int adc_timinit(struct stm32_dev_s *priv)
    *   position.
    */
 
-  ainfo("Initializing timers extsel = 0x%08x\n", priv->extsel);
+  ainfo("Initializing timers extsel = 0x%08lx\n", priv->extsel);
```
[the remark this answers is not in the document]
Comment (robertobucher) [reply arch/arm/src/stm32h7/stm32_adc.c]: Done

**Comment by robertobucher** [comment]
The "pull request" seems ok, but I can't see my new files under boards/arm/stm32h7/nucleo-h743zi2/src:

stm32_pwm.c
stm32_qencoder.c
stm32_gpio.c

and the modified files for pysimCoder

stm32_bringup.c, stm32_appinitialize etc...

**Comment by acassis** [comment]
Hi @robertobucher please squash these 5 commits in a single one. Then you will see all the files. Also please change the PR name to something like: "boards/stm32h7zi2: Add support to PWM, ADC and QE to use with PySimCoder"

**Pull request closed without merging**